test/storage/slab/check_slab.c

```diff
@@ -492,6 +548,80 @@ START_TEST(test_evict_lru_basic)
 }
 END_TEST
 
+START_TEST(test_refcount)
+{
+#define KEY "key"
+#define VAL "val"
+    struct bstring key, val;
+    item_rstatus_t status;
+    struct item *it;
+    struct slab * s;
+    uint32_t dataflag = 12345;
+
+    test_reset();
+
+    key = str2bstr(KEY);
+    val = str2bstr(VAL);
+
+    /* reserve & release */
+    status = item_reserve(&it, &key, &val, val.len, dataflag, 0);
+    ck_assert_msg(status == ITEM_OK, "item_reserve not OK - return status %d", status);
+    s = item_to_slab(it);
+    ck_assert_msg(s->refcount == 1, "slab refcount %"PRIu32"; 1 expected", s->refcount);
+    item_release(&it);
+    ck_assert_msg(s->refcount == 0, "slab refcount %"PRIu32"; 0 expected", s->refcount);
+
+    /* reserve & backfill (link) */
+    status = item_reserve(&it, &key, &val, val.len, dataflag, 0);
+    ck_assert_msg(status == ITEM_OK, "item_reserve not OK - return status %d", status);
+    s = item_to_slab(it);
+    ck_assert_msg(s->refcount == 1, "slab refcount %"PRIu32"; 1 expected", s->refcount);
+    val = null_bstring;
+    item_backfill(it, &val, false);
+    ck_assert_msg(s->refcount == 1, "slab refcount %"PRIu32"; 1 expected", s->refcount);
+    val = null_bstring;
+    item_backfill(it, &val, true);
+    ck_assert_msg(s->refcount == 0, "slab refcount %"PRIu32"; 0 expected", s->refcount);
+}
+END_TEST
+
+START_TEST(test_evict_refcount)
```

Comment:
This test changes the slab config so the second reserve should fail, instead of succeed, (ITEM_ENOMEM expected). And another reserve is attempted after ```item_insert```, which should succeed now that we can evict.